### PR TITLE
chore(deps): bump eth-snap-keyring to 3.0.0

### PR DESCRIPTION
**Pull request by montelaidev**
## **Description**

This PR bumps the `@metamask/eth-snap-keyring` to 3.0.0

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by montelaidev** [comment]
@metamaskbot update-policies

**Review by gantunesr — DISMISSED** [review]

File: yarn.lock [hunk 1]
```diff
@@ -5085,7 +5085,7 @@ __metadata:
   languageName: node
   linkType: hard
 
-"@metamask/eth-snap-keyring@npm:^2.1.1, @metamask/eth-snap-keyring@npm:^2.1.2":
+"@metamask/eth-snap-keyring@npm:^2.1.1":
```
Comment (legobeat) [inline yarn.lock:5088]: Note: The previous v2 is still used by `@metamask/accounts-controller`, which means there are now two instances (dupes) of the package.

Comment (ccharly) [reply yarn.lock:5088]: You mean this, right?
```
└─ metamask-crx@workspace:.
   ├─ @metamask/eth-snap-keyring@npm:3.0.0 (via npm:^3.0.0)
   └─ @metamask/accounts-controller@npm:11.0.0 [15918] (via npm:^11.0.0 [15918])
      └─ @metamask/eth-snap-keyring@npm:2.1.2 (via npm:^2.1.1)
```

Let me check with @montelaidev if we can use the v3, but yes, that should be ok IMO, good catch :)

Comment (ccharly) [reply yarn.lock:5088]: Also, this PR is required for the on-going release, so we cannot align the `@metamask/accounts-controller` now, we might do it later.

Would this be ok for you?

**Comment by montelaidev** [comment]
No longer needed because its bumped in this pr https://github.com/MetaMask/metamask-extension/pull/24130

**Pull request closed without merging**